TaskfMRIAnalysis/scripts/makeSubjectTaskSummary.sh

```diff
@@ -0,0 +1,363 @@
+fi
+
+# Record additional lowpass filter in filenames
+if [ "${TemporalSmoothing}" != "NONE" ]; then
+	LowPassSTRING="_lp""$TemporalSmoothing"
+else
+	LowPassSTRING=""
+fi
+
+# Set variables used for different registration procedures
+if [ "${RegName}" != "NONE" ] ; then
+	RegString="_${RegName}"
+else
+	RegString=""
+fi
+
+if [[ "${SummaryName}" = "NONE" || "${SummaryName}" = "" ]]; then
+	if [ "${LevelTwofMRIName}" = "NONE" ]; then
+	    log_Err_Abort "Cannot determine summaryname. You must provide name for single-subject summary directory if Level2 analysis is not being run."
+	else
+		SummaryName="${LevelTwofMRIName}/${LevelTwofsfName}";
+	fi
+fi
+
+log_Msg "MAIN: SET_NAME_STRINGS: SmoothingString: ${SmoothingString}"
+log_Msg "MAIN: SET_NAME_STRINGS: TemporalFilterString: ${TemporalFilterString}"
+log_Msg "MAIN: SET_NAME_STRINGS: RegString: ${RegString}"
+
+
+# Determine locations of necessary directories (using expected naming convention)
+ResultsFolder="${Path}/${Subject}/MNINonLinear/Results"
+
+# Determine which analyses need to be packaged into summary directory
+# initialize run variables
+runParcellated=false; runVolume=false; runDense=false;
+Analyses=""; ExtensionList=""; ScalarExtensionList="";
+
+# Determine whether to run Parcellated, and set strings used for filenaming
+if [ "${Parcellation}" != "NONE" ] ; then
+	# Run Parcellated Analyses
+	runParcellated=true;
+	ParcellationString="_${Parcellation}"
+	Extension="ptseries.nii"
+	ScalarExtensionList="${ScalarExtensionList}pscalar.nii "
+	Analyses="${Analyses}ParcellatedStats "; # space character at end to separate multiple analyses
+fi
+
+# Determine whether to run Dense, and set strings used for filenaming
+if [ "${Parcellation}" = "NONE" ]; then
+	# Run Dense Analyses
+	runDense=true;
+	ParcellationString=""
+	Extension="dtseries.nii"
+	ExtensionList="${ExtensionList}dtseries.nii "
+	ScalarExtensionList="${ScalarExtensionList}dscalar.nii "
+	Analyses="${Analyses}GrayordinatesStats "; # space character at end to separate multiple analyses
+fi
+
+# Determine whether to run Volume, and set strings used for filenaming
+if [ "$VolumeBasedProcessing" = "YES" ] ; then
+	runVolume=true;
+	ExtensionList="${ExtensionList}nii.gz "
+	ScalarExtensionList="${ScalarExtensionList}volume.dscalar.nii "
+	Analyses="${Analyses}StandardVolumeStats "; # space character at end to separate multiple analyses	
+fi
+
+
+# Determine location where SummaryDirectory should be created
+SummaryDirectory="${ResultsFolder}/${SummaryName}${TemporalFilterString}${SmoothingString}${RegString}${ProcSTRING}${LowPassSTRING}${ParcellationString}_subjectSummary.feat"
+
+# Check if summary directory already exists, and remove old summary directory
+if [ -d "$SummaryDirectory" ]; then
+  echo "Removing previous summary directory from ${SummaryDirectory}..."
+  rm -r "$SummaryDirectory"
+fi
+
+# Create new summary directory
+echo "Creating new summary directory at ${SummaryDirectory}..."
+mkdir -v $SummaryDirectory
```

Comment:
I am puzzled by the mix of quotes and not-quotes in commands.  It is technically more important to use quotes in commands like `mkdir` (allows them to run on folders with spaces, etc in the name) than it is in reporting like `log_Msg` (mainly allows multiple consecutive spaces as-is, and literal use of some shell-special characters), and yet most uses of unquoted variables are in the processing commands where quotes would have more practical benefit, while for `log_Msg` and similar, you virtually always use quotes.

Reply:
I'm not sure I follow. What do you want me to change?

Reply:
`mkdir -v "$SummaryDirectory"`

Using quotes like this in all commands that use a variable (that is not intended to become multiple arguments) would make it whitespace-safe.  We have at least one pipelines author who doesn't care to do this, but since you are using quotes already in lots of other places (including simple variable assignment, where technically they don't matter), why not use them in the most important places (commands)?